### PR TITLE
Adding Shifter implementation

### PR DESCRIPTION
**Pull request by osana**

### CODE REVIEW

**Comment by springmeyer** [comment]
Thank you for this work @osana and the Android team. As https://github.com/mapbox/mapbox-plugin-android-china/pull/174 is depending on this, could I ask for a summary of what remains on this PR before merging and an overall description of what it entails?

**Comment by osana** [comment]
@springmeyer We need to figure out how to make sure
fromJson->toJson->fromJson result in equal objects. The problem is that we are not allowed to `unshift` as we were originally hoping to do. Now we have do make sure that geoJson data has an indication that coordinates were shifted so that they are not shifted again when geoJson data is parsed

**Comment by langsmith** [comment]
@springmeyer

This pr's work adds interface usage to this Java SDK's `Point` GeoJSON class. [The `Point` class now relies on the `CoordinateShifter`'s lat/lng/alt values](https://github.com/mapbox/mapbox-java/pull/922/files#diff-b7d1768ac3665fa886a6bc765eed2ff6R101
).

[The `CoordinateShifter` interface](https://github.com/mapbox/mapbox-java/pull/922/files#diff-2b34568be94c85dd5c747114ed033418) has the `shiftLonLat()` and `shiftLonLatAlt()` methods. As @osana said above, the unshift methods will be removed because Chris didn't want customers to have the ability to unshift points (e.g. by using [Evil Transform](https://github.com/googollee/eviltransform)). 

The `CoordinateShifter` interface allows for someone to create a class which implements `CoordinateShifter` and then override the interface's methods. The actual shifting is handeled inside the overriden methods. You can see this being done downstream in [the `#174` `/mapbox-plugin-android-china` pull request](https://github.com/mapbox/mapbox-plugin-android-china/pull/174/files#diff-feba49855d279275c08d44aa3b64f619R33). With the `ChinaCoordinateShifter` class set up within the plugin, the only remaining line to run before adding data to the map is `CoordinateShifterManager.setCoordinateShifter(ChinaCoordinateShifter())` ([here in the plugin](https://github.com/mapbox/mapbox-plugin-android-china/pull/174/files#diff-ef34078ddb4b9775325e01e0ca4be0d8R22)).

**Comment by langsmith** [comment]
Once #925 lands, let's make sure to rebase this pr so that code coverage reporting gets applied to this work

**Comment by langsmith** [comment]
I've just rebased this branch to account for #925 landing in `master`

File: services-geojson/src/main/java/com/mapbox/geojson/shifter/CoordinateShifterManager.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+  };
+
+  private static volatile CoordinateShifter coordinateShifter = DEFAULT;
+
+  /**
+   * Currently set CoordinateShifterManager.
+   *
+   * @return Currently set CoordinateShifterManager
+   * @since 4.1.0
```
Comment (langsmith) [inline services-geojson/src/main/java/com/mapbox/geojson/shifter/CoordinateShifterManager.java]: We'll need to switch to 4.1.1

File: services-geojson/src/main/java/com/mapbox/geojson/shifter/CoordinateShifterManager.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+   */
+  public static CoordinateShifter getCoordinateShifter() {
+    return coordinateShifter;
+  }
+
+  /**
+   * Sets CoordinateShifterManager.
+   *
+   * @param coordinateShifter CoordinateShifterManager to be set
+   * @since 4.1.0
```
Comment (langsmith) [inline services-geojson/src/main/java/com/mapbox/geojson/shifter/CoordinateShifterManager.java]: We'll need to switch to 4.1.1

File: services-geojson/src/main/java/com/mapbox/geojson/shifter/CoordinateShifterManager.java (hunk 1 above)
Comment (langsmith) [reply services-geojson/src/main/java/com/mapbox/geojson/shifter/CoordinateShifterManager.java]: Did myself

**Comment by langsmith** [comment]
Made some commits to clean up javadoc, which now means that I can't give approval through Github. Looks good to me. Would love some final review from you @LukasPaczos .

As discussed offline with @springmeyer , let's now worry about getting Codecov to accurately report on this particular pr. @osana 's taken the time to write many tests for the work being added on this pr, so we know we've got solid coverage. Once this pr is merged to `master`, we can see how much of this work is uncovered and then open a new pr to add additional tests. Let's not have Codecov funkiness on this pr, block this work entirely.  Ticketed https://github.com/mapbox/mapbox-java/issues/933 so that we don't forget to check for any other needed code coverage.

**Comment by osana** [comment]
@LukasPaczos 
 List instantiation was moved into the shifter so we are not creating more Lists than we did before. Only one is created that is needed for `new AutoValue_Point(TYPE, null, coordinates)`

File: services-geojson/src/main/java/com/mapbox/geojson/shifter/CoordinateShifter.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+import com.mapbox.geojson.Point;
+
+import java.util.List;
+
+/**
+ * ShifterManager allows the movement of all Point objects according to a custom algorithm.
+ * Once set, it will be applied to all Point objects created through this method.
+ *
+ * @since 4.1.2
```
[the remark this answers is not in the document]
Comment (osana) [reply services-geojson/src/main/java/com/mapbox/geojson/shifter/CoordinateShifter.java]: I think you are right! Thank you @LukasPaczos

File: services-geojson/src/test/java/com/mapbox/geojson/shifter/ShifterTest.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    assertEquals(southwestManualShifted, southwestShifted);
+    assertEquals(northeastManualShifted, northeastShifted);
+
+    CoordinateShifterManager.setCoordinateShifter(null);
+  }
+
+  @Test
+  public void bbox_basic_shift() throws Exception {
+
+    CoordinateShifter shifter = new TestCoordinateShifter();
```
[the remark this answers is not in the document]
Comment (osana) [reply services-geojson/src/test/java/com/mapbox/geojson/shifter/ShifterTest.java]: Good catch - updating!

**Comment by springmeyer** [comment]
Thanks for landing this @osana @langsmith @LukasPaczos 🙌

**Pull request merged**